EdgeOfEmpireBot/EdgeOfEmpireBot/MessageHandlers/Router/MessageRouter.cs

```diff
@@ -0,0 +1,69 @@
+using Discord.WebSocket;
+using HK47.MessageHandlers.Interfaces;
+using HK47.Services;
+
+namespace HK47.Router;
+
+public class MessageRouter(IDataService dataService, ISteamMessageHandler steamHandler, IGeneralMessageHandler generalHandler, IMessageService messageService) : IMessageRouter
+{
+    public async Task ProcessMessage(SocketMessage message)
+    {
+        try
+        {
+            Console.WriteLine("Processing Message...");
+            if (!message.Content.StartsWith('.'))
+            {
+                // Ignore message.
+                Console.WriteLine("Ignoring Message....");
+                return;
+            }
+            await messageService.ValidateLength(message.Content);
+
+            // Removes the '.' prefix to indicate a bot command.
+            var messageCommand = message.Content.Remove(0, 1);
+            var messageChannel = message.Channel;
+            await RouteCommand(messageCommand);
+
+            Console.WriteLine(messageCommand + "  command processed. ");
+        }
+        catch (Exception ex)
+        {
+            Console.WriteLine("Error Processing Command");
+            Console.WriteLine(ex.ToString());
+        }
+    }
+
+    /// <summary>
+    /// Processes a Command
+    /// </summary>
+    /// <param name="command"></param>
```

Comment:
Should we have something within the <param></param> to indicate what this actually is ie indicating type or potentially some detail waht its doing